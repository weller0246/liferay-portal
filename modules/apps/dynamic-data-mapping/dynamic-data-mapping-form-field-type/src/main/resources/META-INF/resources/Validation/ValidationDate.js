/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import './ValidationDate.scss';

import {ClayInput} from '@clayui/form';
import React, {useState} from 'react';

import Select from '../Select/Select.es';
import {EVENT_TYPES} from './validationReducer';

const MAX_QUANTITY = 999;
const MIN_QUANTITY = 1;

const customDateLabel = Liferay.Language.get('custom-date');
const daysLabel = Liferay.Language.get('days');
const endsOnLabel = Liferay.Language.get('ends-on');
const minusLabel = Liferay.Language.get('minus');
const monthsLabel = Liferay.Language.get('months');
const plusLabel = Liferay.Language.get('plus');
const responseDateLabel = Liferay.Language.get('response-date');
const startsFromLabel = Liferay.Language.get('starts-from');
const yearsLabel = Liferay.Language.get('years');

const responseDateOption = {
	checked: false,
	label: responseDateLabel,
	name: 'responseDate',
	value: 'responseDate',
};

const operationsOptions = [
	{
		checked: false,
		label: minusLabel,
		name: 'minus',
		value: 'minus',
	},
	{
		checked: false,
		label: plusLabel,
		name: 'plus',
		value: 'plus',
	},
];

const unitOptions = [
	{
		checked: false,
		label: daysLabel,
		name: 'days',
		value: 'days',
	},
	{
		checked: false,
		label: monthsLabel,
		name: 'months',
		value: 'months',
	},
	{
		checked: false,
		label: yearsLabel,
		name: 'years',
		value: 'years',
	},
];

const getDateOptionsByType = (label, name) => ({
	label,
	name,
	options: [
		responseDateOption,
		{
			checked: false,
			label: customDateLabel,
			name: 'customDate',
			value: 'customDate',
		},
	],
});

const getOperation = (quantity) => {
	return quantity < 0 ? 'minus' : 'plus';
};

const getSelectedParameter = (
	localizedValue,
	selectedParameterName,
	attributeName
) => {
	if (localizedValue && typeof localizedValue === 'string') {
		try {
			localizedValue = JSON.parse(localizedValue);
		}
		catch (error) {}
	}

	return localizedValue?.[selectedParameterName]?.[attributeName];
};

const getSignedValue = (operation, value) => {
	let signedValue = value;

	if (
		(operation === 'minus' && value > 0) ||
		(operation === 'plus' && value < 0)
	) {
		signedValue = value * -1;
	}

	return signedValue;
};

const parameters = {
	dateRange: [
		getDateOptionsByType(startsFromLabel, 'startsFrom'),
		getDateOptionsByType(endsOnLabel, 'endsOn'),
	],
	futureDates: [getDateOptionsByType(startsFromLabel, 'startsFrom')],
	pastDates: [getDateOptionsByType(endsOnLabel, 'endsOn')],
};

const CustomDates = ({
	date,
	eventType,
	handleChangeParameters,
	localizationMode,
	operation,
	quantity,
	readOnly,
	setDate,
	setOperation,
	setQuantity,
	setUnit,
	unit,
	visible,
}) => {
	return (
		<>
			<Select
				label={Liferay.Language.get('date')}
				name={`selectedDate_${eventType}`}
				onChange={(event, value) => {
					setDate(value[0]);
					handleChangeParameters(value[0], eventType, 'date');
				}}
				options={[responseDateOption]}
				readOnly={readOnly || localizationMode}
				showEmptyOption={false}
				value={date}
				visible={visible}
			/>
			<div className="align-items-end d-flex position-relative">
				<div className="ddm-form-field-type__validation-date pr-2">
					<Select
						label={Liferay.Language.get('operation')}
						name={`selectedOperation_${eventType}`}
						onChange={(event, value) => {
							setOperation(value[0]);
							handleChangeParameters(
								value[0],
								eventType,
								'quantity'
							);
						}}
						options={operationsOptions}
						readOnly={readOnly || localizationMode}
						showEmptyOption={false}
						value={operation}
						visible={visible}
					/>
				</div>
				<div className="ddm-form-field-type__validation-date pr-2">
					<div className="form-group">
						<label htmlFor="validation_date_quantity">
							{Liferay.Language.get('quantity')}
						</label>
						<ClayInput
							className="ddm-field-text"
							disabled={readOnly}
							id="validation_date_quantity"
							max={MAX_QUANTITY}
							name={`inputedQuantity_${eventType}`}
							onBlur={(event) => {
								let {value: newValue} = event.target;

								if (newValue > MAX_QUANTITY) {
									newValue = MAX_QUANTITY;
								}
								else if (newValue < MIN_QUANTITY) {
									newValue = MIN_QUANTITY;
								}

								newValue =
									operation === 'minus'
										? (newValue * -1).toString()
										: newValue;

								setQuantity(newValue);
								handleChangeParameters(
									newValue,
									eventType,
									'quantity'
								);
							}}
							onChange={({target: {value}}) => {
								setQuantity(value);
							}}
							type="number"
							value={quantity === '' ? '' : Math.abs(quantity)}
						/>
					</div>
				</div>
				<div className="ddm-form-field-type__validation-date">
					<Select
						label={Liferay.Language.get('unit')}
						name={`selectedUnit_${eventType}`}
						onChange={(event, value) => {
							setUnit(value[0]);
							handleChangeParameters(value[0], eventType, 'unit');
						}}
						options={unitOptions}
						readOnly={readOnly || localizationMode}
						showEmptyOption={false}
						value={unit}
						visible={visible}
					/>
				</div>
			</div>
		</>
	);
};

const ValidationDate = ({
	dispatch,
	errorMessage,
	localizationMode,
	localizedValue,
	name,
	onBlur,
	parameter,
	readOnly,
	selectedValidation,
	transformSelectedValidation,
	validations,
	visible,
}) => {
	const initialStartQuantity = getSelectedParameter(
		localizedValue(parameter),
		'startsFrom',
		'quantity'
	);
	const initialEndQuantity = getSelectedParameter(
		localizedValue(parameter),
		'endsOn',
		'quantity'
	);
	const initialStartDate = getSelectedParameter(
		localizedValue(parameter),
		'startsFrom',
		'date'
	);
	const initialEndDate = getSelectedParameter(
		localizedValue(parameter),
		'endsOn',
		'date'
	);

	const initialStartType = getSelectedParameter(
		localizedValue(parameter),
		'startsFrom',
		'type'
	);
	const initialEndType = getSelectedParameter(
		localizedValue(parameter),
		'endsOn',
		'type'
	);
	const initialStartUnit = getSelectedParameter(
		localizedValue(parameter),
		'startsFrom',
		'unit'
	);
	const initialEndUnit = getSelectedParameter(
		localizedValue(parameter),
		'endsOn',
		'unit'
	);

	const selectedParameter = parameters[selectedValidation.name];
	const [startDate, setStartDate] = useState(initialStartDate);
	const [startOperation, setStartOperation] = useState(
		getOperation(initialStartQuantity)
	);
	const [startQuantity, setStartQuantity] = useState(initialStartQuantity);
	const [startsFrom, setStartsFrom] = useState(initialStartType);

	const [startUnit, setStartUnit] = useState(initialStartUnit);

	const [endDate, setEndDate] = useState(initialEndDate);
	const [endOperation, setEndOperation] = useState(
		getOperation(initialEndQuantity)
	);
	const [endQuantity, setEndQuantity] = useState(initialEndQuantity);
	const [endsOn, setEndsOn] = useState(initialEndType);
	const [endUnit, setEndUnit] = useState(initialEndUnit);

	const handleChangeParameters = (value, typeName, type) => {
		const parameter = {};

		let newValue = value;

		if (startsFrom) {
			parameter['startsFrom'] = {
				date: startDate,
				quantity: startQuantity,
				type: startsFrom,
				unit: startUnit,
			};
		}

		if (endsOn) {
			parameter['endsOn'] = {
				date: endDate,
				quantity: endQuantity,
				type: endsOn,
				unit: endUnit,
			};
		}

		let operation =
			typeName === 'startsFrom' ? startOperation : endOperation;

		if (type === 'quantity') {
			let quantity = value;

			if (isNaN(value)) {
				operation = value;
				quantity =
					typeName === 'startsFrom' ? startQuantity : endQuantity;
			}

			newValue = getSignedValue(operation, quantity);
		}

		parameter[typeName] = {
			...parameter[typeName],
			[type]: newValue,
		};

		dispatch({payload: {parameter}, type: EVENT_TYPES.SET_PARAMETER});
	};

	const errorMessageName = name + '_errorMessage';
	const getDateTypeValue = (name) => {
		return name === 'startsFrom' ? startsFrom : endsOn;
	};

	return (
		<>
			<Select
				disableEmptyOption
				label={Liferay.Language.get('accepted-date')}
				name="selectedValidation"
				onChange={(event, value) => {
					setStartsFrom(initialStartType);
					setEndsOn(initialEndType);
					dispatch({
						payload: {
							selectedValidation: transformSelectedValidation(
								value
							),
						},
						type: EVENT_TYPES.CHANGE_SELECTED_VALIDATION,
					});
				}}
				options={validations}
				placeholder={Liferay.Language.get('choose-an-option')}
				readOnly={readOnly || localizationMode}
				showEmptyOption={false}
				value={[selectedValidation.name]}
				visible={visible}
			/>
			{selectedParameter.map((element, index) => {
				let label = '';
				const startSection = element.name === 'startsFrom';

				if (selectedParameter.length > 1) {
					label = startSection
						? Liferay.Language.get('start-date')
						: Liferay.Language.get('end-date');
				}

				return (
					<>
						{label !== '' && (
							<>
								<label>{label.toUpperCase()}</label>
								<div className="separator" />
							</>
						)}

						<Select
							disableEmptyOption
							key={`selectedParameter_${index}`}
							label={element.label}
							name="selectedParameter"
							onChange={(event, value) => {
								if (startSection) {
									setStartsFrom(value[0]);
									setStartOperation(
										getOperation(initialStartQuantity)
									);
									setStartQuantity(initialStartQuantity);
									setStartDate(initialStartDate);
								}
								else {
									setEndsOn(value[0]);
									setEndOperation(
										getOperation(initialEndQuantity)
									);
									setEndQuantity(initialEndQuantity);
									setEndDate(initialEndDate);
								}
								handleChangeParameters(
									value[0],
									element.name,
									'type'
								);
							}}
							options={selectedParameter[index].options}
							placeholder={Liferay.Language.get(
								'choose-an-option'
							)}
							readOnly={localizationMode || readOnly}
							showEmptyOption={false}
							value={startSection ? startsFrom : endsOn}
							visible={visible}
						/>

						{getDateTypeValue(element.name) === 'customDate' && (
							<CustomDates
								date={startSection ? startDate : endDate}
								endDate={endDate}
								eventType={element.name}
								handleChangeParameters={handleChangeParameters}
								operation={
									startSection ? startOperation : endOperation
								}
								quantity={
									startSection ? startQuantity : endQuantity
								}
								readOnly={localizationMode || readOnly}
								setDate={
									startSection ? setStartDate : setEndDate
								}
								setOperation={
									startSection
										? setStartOperation
										: setEndOperation
								}
								setQuantity={
									startSection
										? setStartQuantity
										: setEndQuantity
								}
								setUnit={
									startSection ? setStartUnit : setEndUnit
								}
								unit={startSection ? startUnit : endUnit}
								visible={visible}
							/>
						)}
					</>
				);
			})}
			<label htmlFor={errorMessageName}>
				{Liferay.Language.get('error-message')}
			</label>
			<ClayInput
				id={errorMessageName}
				name={errorMessageName}
				onBlur={onBlur}
				onChange={(event) => {
					dispatch({
						payload: {
							errorMessage: event.target.value,
						},
						type: EVENT_TYPES.CHANGE_ERROR_MESSAGE,
					});
				}}
				placeholder={Liferay.Language.get('this-field-is-invalid')}
				type="text"
				value={localizedValue(errorMessage)}
			/>
		</>
	);
};

export default ValidationDate;
