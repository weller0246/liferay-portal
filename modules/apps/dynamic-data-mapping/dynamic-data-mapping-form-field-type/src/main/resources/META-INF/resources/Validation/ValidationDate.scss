.ddm-form-field-type__validation-date {
	width: 33%;
}
