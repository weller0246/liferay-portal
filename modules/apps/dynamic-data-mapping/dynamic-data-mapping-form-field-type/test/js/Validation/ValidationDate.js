/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '@testing-library/jest-dom/extend-expect';
import {cleanup, render} from '@testing-library/react';
import React from 'react';

import ValidationDate from '../../../src/main/resources/META-INF/resources/Validation/ValidationDate';

const globalLanguageDirection = Liferay.Language.direction;

const validations = [
	{
		checked: false,
		label: 'Future Dates',
		name: 'futureDates',
		parameterMessage: '',
		template: 'futureDates({name}, "{parameter}")',
		value: 'futureDates',
	},
	{
		checked: false,
		label: 'Past Dates',
		name: 'pastDates',
		parameterMessage: '',
		template: 'pastDates({name}, "{parameter}")',
		value: 'pastDates',
	},
	{
		checked: false,
		label: 'Range',
		name: 'dateRange',
		parameterMessage: '',
		template: 'dateRange({name}, "{parameter}")',
		value: 'dateRange',
	},
];

describe('ValidationDate', () => {
	beforeAll(() => {
		Liferay.Language.direction = {
			en_US: 'rtl',
		};
	});

	afterAll(() => {
		Liferay.Language.direction = globalLanguageDirection;
	});

	afterEach(cleanup);

	it('shows future dates validation', () => {
		const {container} = render(
			<ValidationDate
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				localizedValue={() => {}}
				name="validationDate"
				onChange={() => {}}
				selectedValidation={{
					label: '',
					name: 'futureDates',
					parameterMessage: '',
					template: 'futureDates({name}, "{parameter}")',
				}}
				validations={validations}
				visible={true}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('shows past dates validation', () => {
		const {container} = render(
			<ValidationDate
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				localizedValue={() => {}}
				name="validationDate"
				onChange={() => {}}
				selectedValidation={{
					label: '',
					name: 'pastDates',
					parameterMessage: '',
					template: 'pastDates({name}, "{parameter}")',
				}}
				validations={validations}
				visible={true}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('shows date range validation', () => {
		const {container} = render(
			<ValidationDate
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				localizedValue={() => {}}
				name="validationDate"
				onChange={() => {}}
				selectedValidation={{
					label: '',
					name: 'dateRange',
					parameterMessage: '',
					template: 'dateRange({name}, "{parameter}")',
				}}
				validations={validations}
				visible={true}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('shows custom date fields for Future dates', () => {
		const parameter = {
			en_US: {
				startsFrom: {
					date: 'responseDate',
					quantity: 1,
					type: 'customDate',
					unit: 'days',
				},
			},
		};

		const localizedValue = jest.fn(() => parameter['en_US']);
		const {container} = render(
			<ValidationDate
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				localizedValue={localizedValue}
				name="validationDate"
				onChange={() => {}}
				parameter={parameter}
				selectedValidation={{
					label: '',
					name: 'futureDates',
					parameterMessage: '',
					template: 'futureDates({name}, "{parameter}")',
				}}
				validations={validations}
				visible={true}
			/>
		);

		const startDate = container.querySelector(
			'select[name="selectedDate_startsFrom_field"]'
		);

		const startOperation = container.querySelector(
			'select[name="selectedOperation_startsFrom_field"]'
		);

		const startQuantity = container.querySelector(
			'input[name="inputedQuantity_startsFrom"]'
		);

		const startUnit = container.querySelector(
			'input[name="selectedUnit_startsFrom"]'
		);

		expect(startDate).toBeInTheDocument();
		expect(startOperation).toBeInTheDocument();
		expect(startOperation.value).toBe('plus');
		expect(startQuantity).toBeInTheDocument();
		expect(startQuantity.value).toBe('1');
		expect(startUnit).toBeInTheDocument();
		expect(startUnit.value).toBe('days');
	});

	it('shows custom date fields for Past dates and operation minus when quantity is negative', () => {
		const parameter = {
			en_US: {
				endsOn: {
					date: 'responseDate',
					quantity: -1,
					type: 'customDate',
					unit: 'days',
				},
			},
		};

		const localizedValue = jest.fn(() => parameter['en_US']);
		const {container} = render(
			<ValidationDate
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				localizedValue={localizedValue}
				name="validationDate"
				onChange={() => {}}
				parameter={parameter}
				selectedValidation={{
					label: '',
					name: 'pastDates',
					parameterMessage: '',
					template: 'pastDates({name}, "{parameter}")',
				}}
				validations={validations}
				visible={true}
			/>
		);

		const endDate = container.querySelector(
			'select[name="selectedDate_endsOn_field"]'
		);

		const endOperation = container.querySelector(
			'select[name="selectedOperation_endsOn_field"]'
		);

		const endQuantity = container.querySelector(
			'input[name="inputedQuantity_endsOn"]'
		);

		const endUnit = container.querySelector(
			'input[name="selectedUnit_endsOn"]'
		);

		expect(endDate).toBeInTheDocument();
		expect(endOperation).toBeInTheDocument();
		expect(endOperation.value).toBe('minus');
		expect(endQuantity).toBeInTheDocument();
		expect(endUnit).toBeInTheDocument();
	});
});
