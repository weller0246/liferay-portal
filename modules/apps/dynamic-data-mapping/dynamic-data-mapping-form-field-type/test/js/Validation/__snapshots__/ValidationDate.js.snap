// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationDate shows date range validation 1`] = `
<div>
  <div
    class="form-group"
    data-field-name="selectedValidation"
  >
    <label
      class="ddm-label"
      for="selectedValidation"
      tabindex="0"
    >
      accepted-date
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedValidation_field"
      multiple=""
      name="selectedValidation_field"
    >
      <option
        value="dateRange"
      >
        Range
      </option>
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected"
        >
          Range
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedValidation"
      type="hidden"
      value="dateRange"
    />
    <input
      name="selectedValidation_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <label>
    START-DATE
  </label>
  <div
    class="separator"
  />
  <div
    class="form-group"
    data-field-name="selectedParameter"
  >
    <label
      class="ddm-label"
      for="selectedParameter"
      tabindex="0"
    >
      starts-from
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedParameter_field"
      multiple=""
      name="selectedParameter_field"
    >
      <option
        disabled=""
        value=""
      />
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected option-selected-placeholder"
        >
          choose-an-option
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedParameter"
      type="hidden"
      value=""
    />
    <input
      name="selectedParameter_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <label>
    END-DATE
  </label>
  <div
    class="separator"
  />
  <div
    class="form-group"
    data-field-name="selectedParameter"
  >
    <label
      class="ddm-label"
      for="selectedParameter"
      tabindex="0"
    >
      ends-on
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedParameter_field"
      multiple=""
      name="selectedParameter_field"
    >
      <option
        disabled=""
        value=""
      />
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected option-selected-placeholder"
        >
          choose-an-option
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedParameter"
      type="hidden"
      value=""
    />
    <input
      name="selectedParameter_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <label
    for="validationDate_errorMessage"
  >
    error-message
  </label>
  <input
    class="form-control"
    id="validationDate_errorMessage"
    name="validationDate_errorMessage"
    placeholder="this-field-is-invalid"
    type="text"
    value=""
  />
</div>
`;

exports[`ValidationDate shows future dates validation 1`] = `
<div>
  <div
    class="form-group"
    data-field-name="selectedValidation"
  >
    <label
      class="ddm-label"
      for="selectedValidation"
      tabindex="0"
    >
      accepted-date
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedValidation_field"
      multiple=""
      name="selectedValidation_field"
    >
      <option
        value="futureDates"
      >
        Future Dates
      </option>
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected"
        >
          Future Dates
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedValidation"
      type="hidden"
      value="futureDates"
    />
    <input
      name="selectedValidation_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <div
    class="form-group"
    data-field-name="selectedParameter"
  >
    <label
      class="ddm-label"
      for="selectedParameter"
      tabindex="0"
    >
      starts-from
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedParameter_field"
      multiple=""
      name="selectedParameter_field"
    >
      <option
        disabled=""
        value=""
      />
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected option-selected-placeholder"
        >
          choose-an-option
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedParameter"
      type="hidden"
      value=""
    />
    <input
      name="selectedParameter_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <label
    for="validationDate_errorMessage"
  >
    error-message
  </label>
  <input
    class="form-control"
    id="validationDate_errorMessage"
    name="validationDate_errorMessage"
    placeholder="this-field-is-invalid"
    type="text"
    value=""
  />
</div>
`;

exports[`ValidationDate shows past dates validation 1`] = `
<div>
  <div
    class="form-group"
    data-field-name="selectedValidation"
  >
    <label
      class="ddm-label"
      for="selectedValidation"
      tabindex="0"
    >
      accepted-date
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedValidation_field"
      multiple=""
      name="selectedValidation_field"
    >
      <option
        value="pastDates"
      >
        Past Dates
      </option>
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected"
        >
          Past Dates
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedValidation"
      type="hidden"
      value="pastDates"
    />
    <input
      name="selectedValidation_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <div
    class="form-group"
    data-field-name="selectedParameter"
  >
    <label
      class="ddm-label"
      for="selectedParameter"
      tabindex="0"
    >
      ends-on
    </label>
    <select
      aria-hidden="true"
      class="form-control form-control"
      hidden=""
      id="selectedParameter_field"
      multiple=""
      name="selectedParameter_field"
    >
      <option
        disabled=""
        value=""
      />
    </select>
    <div
      class="form-builder-select-field input-group-container"
      style=""
    >
      <div
        class="form-control results-chosen select-field-trigger"
        tabindex="0"
      >
        <div
          class="option-selected option-selected-placeholder"
        >
          choose-an-option
        </div>
        <a
          class="select-arrow-down-container"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-double"
            role="presentation"
          >
            <use
              xlink:href="#caret-double"
            />
          </svg>
        </a>
      </div>
    </div>
    <input
      class="form-control"
      name="selectedParameter"
      type="hidden"
      value=""
    />
    <input
      name="selectedParameter_edited"
      type="hidden"
      value="false"
    />
    
  </div>
  <label
    for="validationDate_errorMessage"
  >
    error-message
  </label>
  <input
    class="form-control"
    id="validationDate_errorMessage"
    name="validationDate_errorMessage"
    placeholder="this-field-is-invalid"
    type="text"
    value=""
  />
</div>
`;
